### PR TITLE
[JENKINS-33600] - User#isIdOrFullnameAllowed() should be tolerant against null parameters

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-33600

@jenkinsci/code-reviewers @reviewbybees @recampbell

### CODE REVIEW

**Comment by jtnord** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
@reviewbybees done

File: core/src/main/java/hudson/model/User.java [hunk 1]
```diff
@@ -702,10 +703,15 @@ private static File getRootDir() {
...
      *
-     * @return true if the username or fullname is valid
+     * @param id ID to be checked
+     * @return {@code true} if the username or fullname is valid.
+     *      For {@code null} or blank IDs returns {@code false}.
      * @since 1.600
      */
-    public static boolean isIdOrFullnameAllowed(String id) {
+    public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
+        if (id == null || StringUtils.isBlank(id)) {
```
Comment (batmat) [inline core/src/main/java/hudson/model/User.java]: `id == null || ` is unneeded, `StringUtils.isBlank(id)` is already documented to return `true` for `null`.

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/User.java]: Unfortunately FindBugs does not know about it.

Comment (batmat) [inline core/src/main/java/hudson/model/User.java]: Granted :-(
IMO tools are great helpers, but we should strive to not change the code just to make them happy.
Maybe disable it on that method, ou use external annotations (not sure it's an existing feature for all IDEs indeed)?

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/User.java]: I'd better keep it as is and let Java optimize it.
But I can remove the check if you feel strongly

**Comment by oleg-nenashev** [comment]
@batmat Are you fine with the current state?

**Comment by batmat** [comment]
I don't know. I know the JVM is gonna optimize and that condition won't have anything noticeable. I'm just worried about readability & maintainability implications in general. Here, I would rather fix FindBugs than the code. But if you like it as-is, I don't feel that strongly, I'll live with it.

**Comment by oleg-nenashev** [comment]
I'll add TODO

**Comment by jtnord** [comment]
> I would rather fix FindBugs than the code

The FindBugs team do not seem to want to accept contributions or work on issues (in my experience)...

**Pull request merged**